blockdag/reachability.go

```diff
@@ -547,9 +811,276 @@ func (dag *BlockDAG) updateReachability(node *blockNode, selectedParentAnticone
+	if i == 0 {
+		return nil, errors.Errorf("rtn is not an ancestor of node")
+	}
+
+	return rootChildrenFutureCoveringSet[i-1].treeNode, nil
+}
+
+func (dag *BlockDAG) concentrateReachabilityTreeIntervalAroundReindexRootChild(
+	reindexRoot *reachabilityTreeNode, chosenReindexRootChild *reachabilityTreeNode) (
+	map[*reachabilityTreeNode]struct{}, error) {
+
+	modifiedTreeNodes := make(map[*reachabilityTreeNode]struct{})
+
+	reindexRootChildNodesBeforeChosen, reindexRootChildNodesAfterChosen, err :=
+		reindexRoot.splitChildrenAroundChosenChild(chosenReindexRootChild)
+	if err != nil {
+		return nil, err
+	}
+
+	reindexRootChildNodesBeforeChosenSizesSum, modifiedNodes, err :=
+		dag.tightenReachabilityTreeIntervalsBeforeChosenReindexRootChild(reindexRoot, reindexRootChildNodesBeforeChosen)
+	if err != nil {
+		return nil, err
+	}
+	for node := range modifiedNodes {
+		modifiedTreeNodes[node] = struct{}{}
+	}
+
+	reindexRootChildNodesAfterChosenSizesSum, modifiedNodes, err :=
+		dag.tightenReachabilityTreeIntervalsAfterChosenReindexRootChild(reindexRoot, reindexRootChildNodesAfterChosen)
+	if err != nil {
+		return nil, err
+	}
+	for node := range modifiedNodes {
+		modifiedTreeNodes[node] = struct{}{}
+	}
+
+	modifiedNodes, err = dag.expandReachabilityTreeIntervalInChosenReindexRootChild(
+		reindexRoot, chosenReindexRootChild, reindexRootChildNodesBeforeChosenSizesSum, reindexRootChildNodesAfterChosenSizesSum)
+	if err != nil {
+		return nil, err
+	}
+	for node := range modifiedNodes {
+		modifiedTreeNodes[node] = struct{}{}
+	}
+
+	return modifiedTreeNodes, nil
+}
+
+func (rtn *reachabilityTreeNode) splitChildrenAroundChosenChild(chosenChild *reachabilityTreeNode) (
+	nodesBeforeChosen []*reachabilityTreeNode, nodesAfterChosen []*reachabilityTreeNode, err error) {
+
+	chosenIndex := -1
+	for i, child := range rtn.children {
+		if child == chosenChild {
+			chosenIndex = i
+			break
+		}
+	}
+	if chosenIndex == -1 {
+		return nil, nil, errors.Errorf("chosenChild not a child of rtn")
+	}
+	return rtn.children[:chosenIndex], rtn.children[chosenIndex+1:], nil
+}
+
+func (dag *BlockDAG) tightenReachabilityTreeIntervalsBeforeChosenReindexRootChild(
+	reindexRoot *reachabilityTreeNode, reindexRootChildNodesBeforeChosen []*reachabilityTreeNode) (
+	reindexRootChildNodesBeforeChosenSizesSum uint64, modifiedTreeNodes map[*reachabilityTreeNode]struct{}, err error) {
+
+	reindexRootChildNodesBeforeChosenSizes, reindexRootChildNodesBeforeChosenSubtreeSizeMaps, reindexRootChildNodesBeforeChosenSizesSum :=
+		calcReachabilityTreeNodeSizes(reindexRootChildNodesBeforeChosen)
+
+	reindexRootStart := reindexRoot.interval.start
+	targetRangeBeforeReindexRootStart := reindexRootStart + reachabilityReindexSlack
+	targetRangeBeforeReindexRootEnd := targetRangeBeforeReindexRootStart + reindexRootChildNodesBeforeChosenSizesSum - 1
+	intervalBeforeReindexRootStart := newReachabilityInterval(targetRangeBeforeReindexRootStart, targetRangeBeforeReindexRootEnd)
```

Comment:
I think the 3 above lines would be much more readable if you do:
```
intervalBeforeReindexRootStart := newReachabilityInterval(
     reindexRootStart + reachabilityReindexSlack,
     targetRangeBeforeReindexRootStart + reindexRootChildNodesBeforeChosenSizesSum - 1)
```

Same for tightenReachabilityTreeIntervalsAfterChosenReindexRootChild